### ISSUE TITLE
Silent failure with runaway caption arguments

### ISSUE THREAD

**Issue by psanan** [shiblon/latex-makefile#8]
I have noticed several silent failures trying to process ill-formed latex files such as the following:

````
\documentclass{article}
\begin{document}
\begin{figure}
\caption{We forgot the closing brace on this caption!
\end{figure}
\end{document}
````

The output from building a makefile and running `make` in a directory with the above tex in `test.tex` is
````
$ make
NOTE: You may ignore warnings about the following files:

     test.d

Makefile:2873: test.d: No such file or directory
= test.tex --> test.d test.pdf.1st.make (0-1) =
= test.tex --> test.d test.pdf.1st.make (1-1) =
= test.tex --> test.pdf (1-2) =
````
Which does not produce any obvious warnings or indications that the make failed and that no pdf was generated (except of course that it doesn't print the usual 'success!' message).

Running pdflatex directly does produce a warning:
````
$ pdflatex test.tex
This is pdfTeX, Version 3.14159265-2.6-1.40.15 (TeX Live 2014/MacPorts 2014_5) (preloaded format=pdflatex)
 restricted \write18 enabled.
entering extended mode
(./test.tex
LaTeX2e <2014/05/01>
Babel <3.9k> and hyphenation patterns for 43 languages loaded.
(/opt/local/share/texmf-texlive/tex/latex/base/article.cls
Document Class: article 2007/10/19 v1.4h Standard LaTeX document class
(/opt/local/share/texmf-texlive/tex/latex/base/size10.clo)) (./test.aux))
Runaway argument?
{We forgot the closing brace on this caption! \end {figure} \end {doc\ETC.
! File ended while scanning use of \@xdblarg.
<inserted text>
                \par
<*> test.tex
````

**Comment by shiblon** [comment]
Huh, that's bad. We should be catching that. That points to a bug in one of the sed scripts that parses the log files.

Can you show the output when running the following in a clean directory?

pdflatex -interaction=batchmode -file-line-error -recorder test.tex

The file-line-error part is important because it changes the way errors are reported, and the makefile makes use of it.

**Comment by shiblon** [comment]
Actually, I just ran it myself (thanks for including the test.tex contents). It looks like file-line-error has *no effect* in this case. No wonder the parser is missing it. I'm looking into a fix now.

**Comment by shiblon** [comment]
OK - I fixed it in commit 0435525910c9705a2f036089f95dcbabe9c5ba43! Please check out the tip and see if this works for you.

**Comment by psanan** [comment]
Fix confirmed on my system!

````
$ make
NOTE: You may ignore warnings about the following files:

     test.d

Makefile:2878: test.d: No such file or directory
= test.tex --> test.d test.pdf.1st.make (0-1) =
= test.tex --> test.d test.pdf.1st.make (1-1) =
Runaway argument?
{We forgot the closing brace on this caption! \end {figure} \end {doc\ETC.
! File ended while scanning use of \@xdblarg.
<inserted text>
                \par
<*> test

I suspect you have forgotten a `}', causing me
to read past where you wanted me to stop.
make: *** [test.pdf] Error 1
````
